### PR TITLE
keystone: add v3 OS-FEDERATION extension List Mappings

### LINKED ISSUE #2461: Add support for identity v3 OS-FEDERATION mappings extension
As part of the `OS-FEDERATION` keystone extension has added a `mappings` api.
Add support for it.

[docs](https://docs.openstack.org/api-ref/identity/v3-ext/index.html#os-federation-api)
[code](https://github.com/openstack/keystone/blob/stable/yoga/keystone/api/os_federation.py#L248)

- [x] List mappings
- [x] Create mapping
- [x] Get mapping
- [x] Update mapping
- [x] Delete mapping

### PR DESCRIPTION
**Pull request by emilmaruszczak**
For #2461

[docs](https://docs.openstack.org/api-ref/identity/v3-ext/index.html#list-mappings)
[code](https://github.com/openstack/keystone/blob/stable/yoga/keystone/api/os_federation.py#L267)

After this is reviewed and merged other operations will follow.

### CODE REVIEW

**Comment by mandre** [comment]
I took the liberty to modify your PR description so that it's not going to close #2461 automatically upon merging this PR, as it appears there are more work needed for it.

**Comment by emilmaruszczak** [comment]
@mandre
I've added the acceptance test, hope it is enought to get this merged.

File: acceptance/openstack/identity/v3/federation_test.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+	client, err := clients.NewIdentityV3Client()
+	th.AssertNoErr(t, err)
+
+  allPages, err := federation.ListMappings(client).AllPages()
+	th.AssertNoErr(t, err)
+
+  mappings, err := federation.ExtractMappings(allPages)
+	th.AssertNoErr(t, err)
+
+	tools.PrintResource(t, mappings)
```
[the remark this answers is not in the document]
Comment (emilmaruszczak) [reply acceptance/openstack/identity/v3/federation_test.go:25]: Yes, as long as `(pagination.LinkedPageBase).ExtractInto` assigns nil as an empty array.

I can create an acceptance test for CRUD (as I've seen in other sets) for create operation and verify that it works as expected. Is it fine with you to leave it as is?

[the remark this answers is not in the document]
Comment (mandre) [reply acceptance/openstack/identity/v3/federation_test.go:25]: That's fine for now, I expect we call again ListMappings when implementing the acceptance tests for the mapping creation.

**Review by mandre — APPROVED** [review]
LGTM

**Pull request merged**